ios/brave-ios/Sources/Brave/Frontend/Browser/Tab.swift

```diff
@@ -391,7 +391,7 @@ class Tab: NSObject {
     $0.addTarget(self, action: #selector(reload), for: .valueChanged)
   }
 
-  func createWebview() {
+  func createWebview(includeDeAmpScript: Bool) {
```

Comment:
This doesn't seem like an ideal solution

Reply:
This is the first time we use a setting that controls a script that comes from brave core:

```
    let scriptPreferences: [UserScriptManager.ScriptType: Bool] = [
        .cookieBlocking: Preferences.Privacy.blockAllCookies.value,
        .mediaBackgroundPlay: Preferences.General.mediaAutoBackgrounding.value,
        .nightMode: Preferences.General.nightModeEnabled.value,
        .deAmp: includeDeAmpScript,
        .playlistMediaSource: Preferences.Playlist.webMediaSourceCompatibility.value,
      ]
```

So far we've been using global prefs but the braveCore is not global. Maybe I need to change `DeAmpPrefs` to a `DeAmpService` and create a factory for it? Then I can access it from the tab as I wish